File: WLC-Assignment/BurgerKing.java

public class BurgerKing {


    public static int calculateMaxProfit(int breads, int vegPattice, 
    int nonVegPattice, int tikkiPattice, int priceVegBurger, 
    int priceNonVegBurger, int priceTikkiBurger) {

        int ans = 0;
        int MaxBurger = (int) Math.floor(breads/2);
        
        for(int i=0; i<MaxBurger; i++) {
        	
        	if ((priceVegBurger >= priceNonVegBurger && priceVegBurger >= priceTikkiBurger) && vegPattice > 0){

                ans += priceVegBurger*vegPattice;
                vegPattice = 0;
                priceVegBurger = 0;

            }else if ((priceNonVegBurger >= priceVegBurger && priceNonVegBurger >= priceTikkiBurger) && nonVegPattice > 0 ){

            	ans += priceNonVegBurger*nonVegPattice;
            	nonVegPattice = 0;
            	priceNonVegBurger = 0;

            }else{
            	
            	ans += priceTikkiBurger*tikkiPattice;
            	tikkiPattice = 0;
            	priceTikkiBurger = 0;
            	
            }
        	
        	
        	
        }
        
     

        return ans;
    
    
    }


   public static void main(String[] args) {
	   
        int breads = 15;
        int vegPattice = 3;
        int nonVegPattice = 2;
        int tikkiPattice = 1;
        int priceVegBurger = 100;
        int priceNonVegBurger = 125;
        int priceTikkiBurger = 112;

        int maxProfit = calculateMaxProfit(breads, vegPattice, nonVegPattice, tikkiPattice, priceVegBurger, priceNonVegBurger, priceTikkiBurger);
        System.out.println("Maximum Profit: $" + maxProfit);

    }

    
}

File: WLC-Assignment/BurgerKing.php
<?php
// Your code here!
/* 

Burgerking sells three items: 
VegBurger which needs 2 breads & 1 veg pattice
NonVegBurger which needs 2 breads & 1 non-veg pattice
TikkiBurger which needs 2 breads & 1 tikki pattice

Given bread quantity, veg pattice quantity, non-veg pattice quantity, tikki pattice quantity & price of all 3 items

Print the total maximum possible profit by making all possible items based on bread availability 

Also, test for all inputs, we would change all the values while testing, the quantity values as well as price

And program has to be optimal with respect to time & space complexity

*/

$breads = 15;
$vegPattice = 3;
$nonVegPattice = 2;
$TikkiPattice = 1;
$priceVegBurger = 100;
$priceNonVegBurger = 125;
$priceTikkiBurger = 112;

$maxProfit = 0;

     // first we divided bread into half by 
    //  that we can know maximum number of burger we can make
     $breads = floor($breads/2);
     
    // if we have breads then we are first going to make as many as non-veg Burger
    // if we dont have sufficient bread then we can make equal no of breads
    
     if($breads > 0){
         
       if($nonVegPattice > 0 && $nonVegPattice <= $breads){
           
         $breads = $breads - $nonVegPattice; 
         $maxProfit += $priceNonVegBurger * $nonVegPattice; 
       
       }else{
       
         $maxProfit += $priceNonVegBurger * $breads; 
        $breads = 0;
       
      }
    }
    
    // in this condition we are counting no of TikkiBurger can be made
    // if we dont have sufficient bread then we can make equal no of breads
    
    if($breads > 0){
        
      if($TikkiPattice > 0 && $TikkiPattice <= $breads){
          
        $breads = $breads - $TikkiPattice;
        $maxProfit += $priceTikkiBurger * $TikkiPattice; 
       
      }else{
   
        $maxProfit +=$priceNonVegBurger * $breads;
        $breads = 0;
       
      }

    }
    
    // in this condition we are counting no of VegBurger can be made
     // if we dont have sufficient bread then we can make equal no of breads
     
    if($breads > 0){

      if($vegPattice > 0 && $vegPattice <= $breads){
        $breads = $breads - $vegPattice;
        $maxProfit += $priceVegBurger * $vegPattice; 
      
      }else{
       
         $maxProfit += $priceNonVegBurger * $breads; 
        $breads = 0;
     
      }
    }
    
    echo $maxProfit

?>

